### PR TITLE
RoleItem added, RoleList modified

### PR DESCRIPTION
**Pull request by emma-martin**
@lauraferrandof could you please review this component? Since this it's instantiated in RolesList, we had to modify it as well. Thanks!!

### CODE REVIEW

File: src/components/RoleList/index.js [hunk 1]
```diff
@@ -9,10 +10,8 @@ class RoleList extends Component {
       <ul className="role-list__list">
         {roles.map((item, index) => {
           return (
-            <li className="role-list__item" key={index}>
-              <div className="role-list__role">{item.role}</div>
-              <div className="role-list__date">{item.from} — {item.until}</div>
-            </li>
+            <RoleItem roleName={item.role} roleDateFrom={item.from} roleDateuntil={item.until} index={index} key={index}/>
```
Comment (sayheylaura) [inline src/components/RoleList/index.js]: the key is necessary where several elements are created. in this case, it's in RoleList, and not in RoleItem, where you do a map and therefore create several items. the key must be in the RoleItem instantiation, and there's no need to pass it to RoleItem through props!

also, I think there's no need either to pass so many props. you can pass {item} as a prop the same way we did in the Experience component. I'll write a code suggestion to illustrate my point!

Comment (sayheylaura) [inline src/components/RoleList/index.js]:
```suggestion
            <RoleItem sampleRoleItem={item} key={index} />
```

File: src/components/RoleItem/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class RoleItem extends Component {
+  render() {
+    const {roleName, roleDateFrom, roleDateuntil, index} = this.props;
```
Comment (sayheylaura) [inline src/components/RoleItem/index.js]:
```suggestion
    const { sampleRoleItem } = this.props;
```

File: src/components/RoleItem/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    return (
+      <li className="role-list__item" key={index}>
```
Comment (sayheylaura) [inline src/components/RoleItem/index.js]:
```suggestion
      <li className="role-list__item">
```

File: src/components/RoleItem/index.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        <p className="role-list__role">{roleName}</p>
```
Comment (sayheylaura) [inline src/components/RoleItem/index.js]:
```suggestion
        <p className="role-list__role">{sampleRoleItem.role}</p>
```

File: src/components/RoleItem/index.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        <p className="role-list__date">{roleDateFrom} — {roleDateuntil}</p>
```
Comment (sayheylaura) [inline src/components/RoleItem/index.js]:
```suggestion
        <p className="role-list__date">{sampleRoleItem.from} — {sampleRoleItem.until}</p>
```

File: src/components/RoleItem/index.js (hunk 2 above)
Comment (sayheylaura) [reply src/components/RoleItem/index.js]: you could even go further in the destructuring:

```suggestion
    const { role, from, until } = this.props.sampleRoleItem;
```
and then just write those below

**Review by sayheylaura — CHANGES_REQUESTED** [review]
@emma-martin good job! Although I believe there are some modifications needed. Start reading in RoleList.js, line 13 :)

File: src/components/RoleItem/index.js (hunk 2 above)
Comment (emma-martin) [reply src/components/RoleItem/index.js]: @lauraferrandof thanks so much for your input! Really liked the simplicity of these props. Tried to go with the deeper destructuring option, but there is a warning regarding propTypes:
"Warning: Failed prop type: The prop `role` is marked as required in `RoleItem`, but its value is `undefined`. in RoleItem (at RoleList/index.js:13)"

I made the following amendments as suggested:
At RoleList `<RoleItem sampleRoleItem={item} key={index}/>`
At RoleItem 
```
const { role, from, until } = this.props.sampleRoleItem;
    console.log(role);
    return (
      <li className="role-list__item">
        <p className="role-list__role">{role}</p>
        <p className="role-list__date">{from} — {until}</p>
      </li>
    );
```
I checked the logs and the props are being received, not sure what I am missing here. Also I double checked our sample and it seems fine. 
I'm pushing the changes for you to review. Thanks a lot!

Comment (sayheylaura) [reply src/components/RoleItem/index.js]: @emma-martin ok, almost there! The problem is RoleItem is receiving ONLY 1 PROP, `sampleRoleItem`. After we destructure this into 3 constants, `role`, `from` and `until`, but these are not strictly props. That's why you're having trouble in your PropTypes. You just have to remove those 3 from PropTypes and leave only `sampleRoleItem: PropTypes.object.isRequired` instead :)

Or if it's less confusing, just leave the first simpler destructuring option! Whatever you feel more comfortable with, the other one was just a suggestion :)

Comment (emma-martin) [reply src/components/RoleItem/index.js]: @lauraferrandof , awesome!!! Ready to merge, thanks so much :-D

**Review by sayheylaura — APPROVED** [review]

**Comment by sayheylaura** [comment]
@emma-martin good job, ready to merge! :)

**Pull request merged**